gravitee-am-repository/gravitee-am-repository-tests/src/test/java/io/gravitee/am/repository/management/api/UserRepositoryTest.java

```diff
@@ -565,6 +565,92 @@ public void testScimSearch_byDate_paged() {
+
+        FilterCriteria criteriaLastPassReset = new FilterCriteria();
+        criteriaLastPassReset.setFilterName("meta.lastPasswordReset");
+        criteriaLastPassReset.setFilterValue(UTC_FORMATTER.format(LocalDateTime.now(UTC).minusSeconds(20)));
+        criteriaLastPassReset.setOperator("gt");
+        TestObserver<Page<User>> testObserverLastPassReset = userRepository.search(ReferenceType.DOMAIN, domain, criteriaLastPassReset, 0, 1).test();
+        testObserverLastPassReset.awaitTerminalEvent();
+
+        testObserverLastPassReset.assertComplete();
+        testObserverLastPassReset.assertNoErrors();
+        testObserverLastPassReset.assertValue(users -> users.getData().size() == 1);
+
+        FilterCriteria criteriaLastMFAEnrollSkippedAt = new FilterCriteria();
+        criteriaLastMFAEnrollSkippedAt.setFilterName("meta.mfaEnrollmentSkippedAt");
+        criteriaLastMFAEnrollSkippedAt.setFilterValue(UTC_FORMATTER.format(LocalDateTime.now(UTC).minusSeconds(20)));
+        criteriaLastMFAEnrollSkippedAt.setOperator("gt");
+        TestObserver<Page<User>> testObserverMFAEnrollSkippedAt = userRepository.search(ReferenceType.DOMAIN, domain, criteriaLastMFAEnrollSkippedAt, 0, 1).test();
+        testObserverMFAEnrollSkippedAt.awaitTerminalEvent();
+
+        testObserverMFAEnrollSkippedAt.assertComplete();
+        testObserverMFAEnrollSkippedAt.assertNoErrors();
+        testObserverMFAEnrollSkippedAt.assertValue(users -> users.getData().size() == 1);
+
+        FilterCriteria criteriaLastAccountLockedAt = new FilterCriteria();
+        criteriaLastAccountLockedAt.setFilterName("meta.accountLockedAt");
+        criteriaLastAccountLockedAt.setFilterValue(UTC_FORMATTER.format(LocalDateTime.now(UTC).minusSeconds(20)));
+        criteriaLastAccountLockedAt.setOperator("gt");
+        TestObserver<Page<User>> testObserverAccLockedAt = userRepository.search(ReferenceType.DOMAIN, domain, criteriaLastAccountLockedAt, 0, 1).test();
+        testObserverAccLockedAt.awaitTerminalEvent();
+
+        testObserverAccLockedAt.assertComplete();
+        testObserverAccLockedAt.assertNoErrors();
+        testObserverAccLockedAt.assertValue(users -> users.getData().size() == 1);
+
+
+        FilterCriteria criteriaLastAccountLockedUntil = new FilterCriteria();
+        criteriaLastAccountLockedUntil.setFilterName("meta.accountLockedUntil");
+        criteriaLastAccountLockedUntil.setFilterValue(UTC_FORMATTER.format(LocalDateTime.now(UTC).minusSeconds(20)));
+        criteriaLastAccountLockedUntil.setOperator("gt");
+        TestObserver<Page<User>> testObserverAccLockedUntil = userRepository.search(ReferenceType.DOMAIN, domain, criteriaLastAccountLockedUntil, 0, 1).test();
+        testObserverAccLockedUntil.awaitTerminalEvent();
+
+        testObserverAccLockedUntil.assertComplete();
+        testObserverAccLockedUntil.assertNoErrors();
+        testObserverAccLockedUntil.assertValue(users -> users.getData().size() == 1);
+    }
+
```

Comment:
**suggestion**: as all the date are initialized using NOW, we may miss a fieldname mapping issue. I would recommend a dedicated test case per date field and initialize all the date field to NOW minus 1H excepted for the field we are looking into. wdyt ?

Reply:
test updated